flink-staging/flink-ml/src/test/scala/org/apache/flink/ml/MLUtilsSuite.scala

```diff
@@ -109,4 +109,41 @@ class MLUtilsSuite extends FlatSpec with Matchers with FlinkTestBase {
 
     tempFile.delete()
   }
+
+  it should "create a discrete histogram" in {
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val discreteData = Seq(1.0, 2.0, 3.0, 5.0, 1.0, 7.0, 9.0, 1.0, 0.0, 1.0, 4.0, 6.0, 7.0, 9.0,
+      4.0, 3.0, 1.0, 4.0, 6.0, 8.0, 4.0, 3.0, 6.0, 8.0, 4.0, 3.0, 6.0, 8.0, 9.0, 7.0, 8.0, 2.0, 3.0,
+      6.0, 0.0)
+    val h = env.fromCollection(discreteData)
+      .createHistogram(0)
+      .collect().toArray.apply(0)
+      .asInstanceOf[CategoricalHistogram]
+
+    h.count(0) should equal(discreteData.filter(x => x == 0).size)
+    h.count(1) should equal(discreteData.filter(x => x == 1).size)
+    h.count(2) should equal(discreteData.filter(x => x == 2).size)
+    h.count(3) should equal(discreteData.filter(x => x == 3).size)
+    h.count(4) should equal(discreteData.filter(x => x == 4).size)
+    h.count(5) should equal(discreteData.filter(x => x == 5).size)
+    h.count(6) should equal(discreteData.filter(x => x == 6).size)
+    h.count(7) should equal(discreteData.filter(x => x == 7).size)
+    h.count(8) should equal(discreteData.filter(x => x == 8).size)
+    h.count(9) should equal(discreteData.filter(x => x == 9).size)
+  }
+
+  it should "create a continuous histogram" in {
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val discreteData = Seq(1.0, 2.0, 3.0, 5.0, 1.0, 7.0, 9.0, 1.0, 0.0, 1.0, 4.0, 6.0, 7.0, 9.0,
+      4.0, 3.0, 1.0, 4.0, 6.0, 8.0, 4.0, 3.0, 6.0, 8.0, 4.0, 3.0, 6.0, 8.0, 9.0, 7.0, 8.0, 2.0, 3.0,
+      6.0, 0.0)
+    val h = env.fromCollection(discreteData)
+      .createHistogram(5)
+      .collect().toArray.apply(0)
+      .asInstanceOf[ContinuousHistogram]
```

Comment:
Ideally we would like to avoid having to cast, and just return the correct type, depending on the provided arguments. Or always return a ContinuousHistogram as I originally suggested.